### ISSUE TITLE
API Documentation

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot-docs#1877]
I find using the API Documentation hard,  because when looking at  a class' methods and attributes, it is not easy to view the inherited ones.  When I click on an inherited class, it takes me to the parent  class. So I lose continuity in trying to figure out which properties/methods to use. It is really hard to follow when jumping around pages.

In either case a class' methods/properties represent everything all the way up to Object. 

The correct way to do API documentation is to provide the ability to expand a parent class summary methods and properties, inside the current page. Not navigate you away from the page. 

In some respects, it is similar to the Godot Editor, that shows the inheritance properties and signals.

So, given that, does the C code have the ability to autogenerate GDScript API docs which shows all methods/properties that the class responds to?

**Comment by Calinou** [comment]
> So, given that, does the C code have the ability to autogenerate GDScript API docs which shows all methods/properties that the class responds to?

That can surely be done, the main issue is providing an UI that actually works well for showing this information.

The same goes for the Web interface: we have to work within Sphinx and reStructuredText's limitations.

**Comment by wyattbiker** [comment]
Will GDQuest see these requests? I heard he will be working on the docs.

**Comment by Calinou** [comment]
@wyattbiker His focus on [redesigning the documentation](https://github.com/godotengine/godot-docs/issues/3390) is mainly about the documentation pages, not the class reference.
